zcash_client_sqlite/src/testing.rs

```diff
@@ -1938,22 +1937,62 @@ impl BlockCache {
     }
 }
 
+pub(crate) struct NoteCommitments {
+    sapling_note_commitments: Vec<sapling::Node>,
+    #[cfg(feature = "orchard")]
+    orchard_note_commitments: Vec<MerkleHashOrchard>,
+}
+
+impl NoteCommitments {
+    #[allow(dead_code)]
+    pub(crate) fn sapling_note_commitments(&self) -> &[sapling::Node] {
+        self.sapling_note_commitments.as_ref()
+    }
+
+    #[cfg(feature = "orchard")]
+    pub(crate) fn orchard_note_commitments(&self) -> &[MerkleHashOrchard] {
+        self.orchard_note_commitments.as_ref()
+    }
+}
+
 impl TestCache for BlockCache {
     type BlockSource = BlockDb;
-    type InsertResult = ();
+    type InsertResult = NoteCommitments;
```

Comment:
Why are we only returning `NoteCommitments` from `impl TestCache for BlockCache`, and not `impl TestCache for FsBlockCache`?

Also, why is this even necessary? `TestCache::insert` takes a `&CompactBlock`, so the caller still has the block and can fetch these commitments from it directly (e.g. with the `NoteCommitments::from_compact_block` constructor I suggest below, or by us adding a helper method to `CompactBlock` in `zcash_client_backend/src/proto.rs`).

Reply:
`InsertResult` for `FsBlockCache` was already returning `BlockMeta` and I saw no reason to change it, because all of the relevant tests are just using `BlockCache`.